Sources/Sentry/SentryUIEventTracker.m

```diff
@@ -0,0 +1,132 @@
+
+                NSString *operation = [self getOperation:event];
+                BOOL sameOperation =
+                    [self.activeTransaction.context.operation isEqualToString:operation];
+
+                if (sameView && sameOperation) {
+                    [self.activeTransaction dispatchIdleTimeout];
+                    return;
+                }
+
+                [self.activeTransaction finish];
+
+                NSString *transactionName = [self getTransactionName:action
+                                                              target:target
+                                                                view:view];
+
+                SentryTransactionContext *context =
+                    [[SentryTransactionContext alloc] initWithName:transactionName
+                                                         operation:operation];
+
+                BOOL ongoingScreenLoadTransaction
+                    = span != nil && [span.context.operation isEqualToString:@"ui.load"];
+                BOOL ongoingManualTransaction = span != nil
+                    && ![span.context.operation isEqualToString:@"ui.load"]
+                    && ![span.context.operation containsString:@"ui.action."];
+
+                BOOL bindToScope = !ongoingScreenLoadTransaction && !ongoingManualTransaction;
+                SentryTracer *transaction =
+                    [SentrySDK.currentHub startTransactionWithContext:context
+                                                          bindToScope:bindToScope
+                                                customSamplingContext:@{}
+                                                          idleTimeout:defaultIdleTransactionTimeout
+                                                 dispatchQueueWrapper:self.dispatchQueueWrapper];
+
+                transaction.finishCallback = ^(void) {
+                    self.activeTransaction = nil;
+                    self.activeView = nil;
+                };
+
+                self.activeTransaction = transaction;
+                self.activeView = view;
+                self.activeEventType = event.type;
+            }];
+        }
+                   forKey:SentryUIEventTrackerSwizzleSendAction];
+}
+
+- (void)stop
+{
+    [self.swizzleWrapper removeSwizzleSendActionForKey:SentryUIEventTrackerSwizzleSendAction];
+}
+
+- (NSString *)getOperation:(UIEvent *)event
+{
+    NSString *operation = @"ui.action";
+    if (event && (event.type == UIEventTypePresses || event.type == UIEventTypeTouches)) {
+        operation = @"ui.action.click";
+    }
+
+    return operation;
+}
+
+- (NSString *)getTransactionName:(NSString *)action target:(id)target view:(UIView *)element
+{
+    NSString *viewIdentifier = action;
+    if (element.accessibilityIdentifier) {
+        viewIdentifier = element.accessibilityIdentifier;
+    }
+
+    Class targetClass = [target class];
+    return [NSString stringWithFormat:@"%@.%@", NSStringFromClass(targetClass), viewIdentifier];
```

Comment:
m: I think we are loosing an important information here. The `action` is the method the was called, the exactly code that run and was captured in the transaction. I think this is way more important than the control identifier. It is possible to change the control action programmatically, and then the user doesn't now which code was traced.

Reply:
Since this is outdated, I will create a new comment in the new approach.